homeassistant/components/motioneye/media_source.py

```diff
@@ -0,0 +1,349 @@
+        cls, config: ConfigEntry, device: dr.DeviceEntry
+    ) -> int:
+        """Get a config entry from a URL."""
+        for identifier in device.identifiers:
+            data = split_motioneye_device_identifier(identifier)
+            if data is not None:
+                return data[2]
+        raise MediaSourceError(f"Could not find camera id for device id: {device.id}")
+
+    @classmethod
+    def _build_media_config(cls, config: ConfigEntry) -> BrowseMediaSource:
+        return BrowseMediaSource(
+            domain=DOMAIN,
+            identifier=config.entry_id,
+            media_class=MEDIA_CLASS_DIRECTORY,
+            media_content_type="",
+            title=config.title,
+            can_play=False,
+            can_expand=True,
+            children_media_class=MEDIA_CLASS_DIRECTORY,
+        )
+
+    def _build_media_configs(self) -> BrowseMediaSource:
+        """Build the media sources for config entries."""
+        return BrowseMediaSource(
+            domain=DOMAIN,
+            identifier="",
+            media_class=MEDIA_CLASS_DIRECTORY,
+            media_content_type="",
+            title="motionEye Media",
+            can_play=False,
+            can_expand=True,
+            children=[
+                self._build_media_config(entry)
+                for entry in self.hass.config_entries.async_entries(DOMAIN)
+            ],
+            children_media_class=MEDIA_CLASS_DIRECTORY,
+        )
+
+    @classmethod
+    def _build_media_device(
+        cls,
+        config: ConfigEntry,
+        device: dr.DeviceEntry,
+        full_title: bool = True,
+    ) -> BrowseMediaSource:
+        return BrowseMediaSource(
+            domain=DOMAIN,
+            identifier=f"{config.entry_id}#{device.id}",
+            media_class=MEDIA_CLASS_DIRECTORY,
+            media_content_type="",
+            title=f"{config.title} {device.name}" if full_title else device.name,
+            can_play=False,
+            can_expand=True,
+            children_media_class=MEDIA_CLASS_DIRECTORY,
+        )
+
+    def _build_media_devices(self, config: ConfigEntry) -> BrowseMediaSource:
+        """Build the media sources for device entries."""
+        device_registry = dr.async_get(self.hass)
+        devices = dr.async_entries_for_config_entry(device_registry, config.entry_id)
+
+        base = self._build_media_config(config)
+        base.children = [
+            self._build_media_device(config, device, full_title=False)
+            for device in devices
+        ]
+        return base
+
+    @classmethod
+    def _build_media_kind(
+        cls,
+        config: ConfigEntry,
+        device: dr.DeviceEntry,
+        kind: str,
+        full_title: bool = True,
+    ) -> BrowseMediaSource:
+        return BrowseMediaSource(
+            domain=DOMAIN,
+            identifier=f"{config.entry_id}#{device.id}#{kind}",
+            media_class=MEDIA_CLASS_DIRECTORY,
+            media_content_type=MEDIA_CLASS_DIRECTORY,
```

Comment:
We're not allowed to use a media class constant for the media content type. The media content type is not the same interface.